torchrec/distributed/tests/test_train_pipeline.py

```diff
@@ -138,15 +138,17 @@ def test_equal_to_non_pipelined(self) -> None:
             for b in range(5)
         ]
         dataloader = iter(data)
-        pipeline = TrainPipelineBase(model_gpu, optimizer_gpu, self.device)
+        pipeline = iter(
```

Comment:
let's add a parallel test for both the new interface and maintain old one (have a lot of internal code to migrate :))

I ran into some tricky StopIteration issues when doing
for batch in pipeline:
   ....

instead of next(pipeline),

but maybe i did something weird

Reply:
I'll make parallel unit test.
you don't need to migrate existing code - keep using progress() for a while.